src/components/graph/graph.builder.js

```diff
@@ -52,13 +52,16 @@ function _getNodeOpacity(node, highlightedNode, highlightedLink, config) {
  * @memberof Graph/builder
  */
 function buildLinkProps(link, nodes, links, config, linkCallbacks, highlightedNode, highlightedLink, transform) {
-    const { source, target } = link;
+    const { source, target, linktype } = link;
     const x1 = nodes?.[source]?.x || 0;
     const y1 = nodes?.[source]?.y || 0;
     const x2 = nodes?.[target]?.x || 0;
     const y2 = nodes?.[target]?.y || 0;
 
-    const d = buildLinkPathDefinition({ source: { x: x1, y: y1 }, target: { x: x2, y: y2 } }, config.link.type);
+    const d = buildLinkPathDefinition(
+        { source: { x: x1, y: y1 }, target: { x: x2, y: y2 }, linktype },
+        config.link.type
```

Comment:
The type is already being passed to `buildLinkPathDefinition`, there's no need to duplicate the paramenter. I would go with:

```javascript
const type = linkType || config.link.type;
const d = buildLinkPathDefinition(
        { source: { x: x1, y: y1 }, target: { x: x2, y: y2 }, linktype },
        type,
);
```

Reply:
o well i over did it  i kept thinking that type argument is global scoped now i realized what was wrong haha thanks a lot